auth/auth0/src/authenticate.js

```diff
@@ -0,0 +1,101 @@
+const isomorphicFetch = require('isomorphic-fetch');
+const jwt = require('jsonwebtoken');
+const jwkRsa = require('jwks-rsa');
+const fromEvent = require('graphcool-lib').fromEvent;
+
+//Validates the request JWT token
+const verifyToken = token =>
+  new Promise(resolve => {
+    //Decode the JWT Token
+    const decoded = jwt.decode(token, { complete: true });
+    if (!decoded || !decoded.header || !decoded.header.kid) {
+      throw new Error('Unable to retrieve key identifier from token');
+    } 
+    const jkwsClient = jwkRsa({
+      cache: true,
+      jwksUri: `https://${process.env.AUTH0_DOMAIN}/.well-known/jwks.json`
+    });
+     //Retrieve the JKWS's signing key using the decode token's key identifier (kid)
+    jkwsClient.getSigningKey(decoded.header.kid, (err, key) => {
+      if (err) throw new Error(err);
+      const signingKey = key.publicKey || key.rsaPublicKey;
+      //If the JWT Token was valid, verify its validity against the JKWS's signing key
+      jwt.verify(
```

Comment:
According to https://auth0.com/docs/tokens/id-token#validate-the-claims, as a minimum you need to check exp, iss and aud.